Payroll list my payments

sgobotta opened the pull request:
This PR provides:

+ payments list in the My Payroll section
+ token and date filters for payments
+ Closes #4

sgobotta committed:
TokenFilter: Creates a basic filter component

sgobotta committed:
DateRangeFilter: implements a filter component for date range

sgobotta committed:
DateRange Input Component: implements a component to choose a date range

sgobotta committed:
Input Components: adds a DateRangeInput component

sgobotta committed:
Utils: adds a math utils module

sgobotta committed:
Formatting Utils: adds a format token amount function

sgobotta committed:
Table Update: the table now passes the item to the render function.

+ When children need to be rendered, it is possible that some props need to be passed. If the table allows the iterating parameter 'item' to the render function, all props sent as 'data' are available to be used and render the desired component properly

sgobotta committed:
Types: declares the structure of a salary

sgobotta committed:
Defines the structure and functionality of a Salary Table

sgobotta committed:
Previous Salary section: implements a first approach of the Previous Salary section in My Payroll

sgobotta committed:
MyPayroll: adds the Previous Salary section to MyPayroll screen

sgobotta committed:
Fixes Date and Amount prop types

sgobotta committed:
Fixes warnings coming from the Table component

sgobotta committed:
TokenFilter: implements the filter using options from the store

sgobotta committed:
Merges payroll-request-payment into payroll-list-my-payments

sgobotta committed:
Store/Marshalling: implements a function that creates a payment

sgobotta committed:
Store: updates the onSendPayRoll function to store payments for an account

sgobotta committed:
Salary Table: refactors the table to receive formatting functions

sgobotta committed:
PreviousSalary section: gets the payments from the store for the Salary Table

sgobotta committed:
Payroll contract: updates SendPayroll event to send the payment date and the exchange rate

sgobotta committed:
Salary Table: Adds the token symbol in the columns, chantes the way the exchange rate is formatted

sgobotta committed:
Marshalling payment: updates the exchangeRate and the payment date values

sgobotta committed:
Init script: Fixes the payment token validation

sgobotta committed:
Types: updates the salary type object

sgobotta committed:
Previous Salary section: removes fake data array

sgobotta committed:
Fixes Previous Salary Table styles

sgobotta committed:
Previous Salary Filters: fixes data range values handling and token filter validation

sgobotta committed:
Refactored styles with styled components

sgobotta committed:
Merges payroll-milestone-3 into payroll-list-my-payments

sgobotta committed:
Merge remote-tracking branch 'origin/payroll-milestone-3' into payroll-list-my-payments-merge-request-payment

josx committed:
Previous Salary test: is it showing title

Signed-off-by: José Luis Di Biase <josx@interorganic.com.ar>

lmcorbalan commented on future-apps/payroll/app/src/screens/components/SalaryTable.js:
```diff
@@ -0,0 +1,91 @@
+      value: data => data.transactionAddress,
+      defaultValue: 'Active',
+      render: (formattedValue, rawValue, item) => (
+        <TransactionAddress>
```
We could use aragon-ui `IdentityBadge` component here, it's the one being used in Finance app

sgobotta replied:
We're not using IdentityBadge but we're planning on create a custom component using SafeLink. The approach has been pushed.

lmcorbalan commented on future-apps/payroll/app/src/store/marshalling.js:
```diff
@@ -34,3 +34,25 @@ export function tokenAllocation (data) {
+        address: data.token.address,
+        symbol: data.token.symbol
+      },
+      decimals: data.token.decimals
```
I would put this inside token to maintain consistency, like:
```
token: {
  address: data.token.address,
  symbol: data.token.symbol,
  decimals: data.token.decimals
},
```

sgobotta replied:
Fixed the attribute

josx committed:
Merge branch 'payroll-milestone-3' into payroll-list-my-payments

sgobotta committed:
Fixes Table rendering

+ Adds a Transaction Link to the hash
+ Fixes the token decimals attribute. Refactored the formatCurrency function

josx committed:
Previously Salary: Fix amount formatting

Signed-off-by: José Luis Di Biase <josx@interorganic.com.ar>
